application/src/main/java/com/sanction/thunder/resources/VerificationResource.java

```diff
@@ -0,0 +1,146 @@
+package com.sanction.thunder.resources;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.sanction.thunder.authentication.Key;
+import com.sanction.thunder.dao.DatabaseError;
+import com.sanction.thunder.dao.DatabaseException;
+import com.sanction.thunder.dao.PilotUsersDao;
+import com.sanction.thunder.models.Email;
+import com.sanction.thunder.models.PilotUser;
+
+import io.dropwizard.auth.Auth;
+
+import javax.inject.Inject;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/verify")
+@Produces(MediaType.APPLICATION_JSON)
+public class VerificationResource {
+  private static final Logger LOG = LoggerFactory.getLogger(UserResource.class);
+
+  private final PilotUsersDao usersDao;
+
+  // Counts number of requests
+  private final Meter postRequests;
+
+  /**
+   * Constructs a new VerificationResource to allow verification of a user.
+   *
+   * @param usersDao The DAO to connect to the database with.
+   * @param metrics The metrics object to set up meters with.
+   */
+  @Inject
+  public VerificationResource(PilotUsersDao usersDao, MetricRegistry metrics) {
+    this.usersDao = usersDao;
+
+    // Set up metrics
+    this.postRequests = metrics.meter(MetricRegistry.name(
+        UserResource.class,
+        "post-requests"));
+  }
+
+  /**
+   * Verifies the provided email, setting it as valid in the database.
+   *
+   * @param key The basic authentication key necessary to access the resource.
+   * @param email The email to verify in the database.
+   * @return A response status and message.
+   */
+  @GET
+  public Response verifyEmail(@Auth Key key,
+                              @QueryParam("email") String email,
+                              @QueryParam("token") String token) {
+    postRequests.mark();
+
+    if (email == null || email.isEmpty()) {
+      LOG.warn("Attempted email verification with an email.");
+      return Response.status(Response.Status.BAD_REQUEST)
+          .entity("Incorrect or missing query parameter.").build();
```

Comment:
Can you make this message more descriptive? Say something about the missing email address?